ui/page/transaction/transactions_page.go

```diff
@@ -371,35 +506,58 @@ func (pg *TransactionsPage) HandleUserInteractions() {
 		break
 	}
 
+	if pg.isHomepageLayout && pg.walletDropDown != nil && pg.walletDropDown.Changed() {
```

Comment:
As long as there's a dropdown, use it. Whether it's home page layout or not. If there should only be a dropdown for homepage layout, then make sure it is nil if it is not home page layout.
```suggestion
	if pg.walletDropDown != nil && pg.walletDropDown.Changed() {
```